pkg/tests/apis/folder/folders_test.go

```diff
@@ -926,3 +926,134 @@ func testDescription(description string, expectedErr error) string {
 		return description
 	}
 }
+
+// TestFoldersGetAPIEndpointK8S is the counterpart of pkg/api/folder_test.go TestFolderGetAPIEndpoint
+func TestFoldersGetAPIEndpointK8S(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	type testCase struct {
+		description    string
+		expectedCode   int
+		params         string
+		createFolders  []string
+		expectedOutput []dtos.FolderSearchHit
+		permissions    []resourcepermissions.SetResourcePermissionCommand
+	}
+
+	folderCreatePermission := []resourcepermissions.SetResourcePermissionCommand{
+		{
+			Actions:           []string{"folders:create"},
+			Resource:          "folders",
+			ResourceAttribute: "uid",
+			ResourceID:        "*",
+		},
+	}
+
+	folder1 := "{ \"uid\": \"foo\", \"title\": \"Folder 1\"}"
+	folder2 := "{ \"uid\": \"bar\", \"title\": \"Folder 2\", \"parentUid\": \"foo\"}"
+	folder3 := "{ \"uid\": \"qux\", \"title\": \"Folder 3\"}"
+
+	tcs := []testCase{
```

Comment:
Looks like we are missing a test case for this: https://github.com/grafana/grafana/blob/main/pkg/api/folder_test.go#L452

"get folder by UID should return parent folders redacted if nested folder are enabled and user does not have read access to parent folders"

Reply:
Actually this test case refers to the `GetFolderByUID` endpoint, not the `GetFolders` endpoint which is the subject of this PR.

In fact, I realized now that there are no tests for the legacy `GetFolders` endpoijnt.
I am going to remove the comment that says that this new test is a counterpart of legacy.

Let me know if this makes sense.

Reply:
Makes a lot of sense. I only see tests for the happy path though. Is there anything we can add relating to permissions? For instance, when would we expect non-200 status codes? For example, one scenario I can think of testing is the following. What happens to the list response if there's an issue with one of the folders (for example converting it to legacy folder DTO)?

Reply:
Hey @suntala , I have added two new testcases:
- listing folders of a parent that does not exists: 404-Not Found
- listing folders from another org: 403-Forbidden